netdata/netdata
# Detect a buggy Ubuntu kernel

##### Summary
There is a buggy Ubuntu HWE kernel (Ubuntu 18.04 - 4.18.0-13) that crashes when it tries to load BPF programs for the eBPF plugin.
```
 kernel BUG at /build/linux-hwe-jDoxDp/linux-hwe-4.18.0/mm/usercopy.c:100!
  35 Jul 30 06:30:56 userv1 kernel: [19816692.905270] invalid opcode: 0000 [#1] SMP NOPTI
  36 Jul 30 06:30:56 userv1 kernel: [19816692.905283] CPU: 8 PID: 5370 Comm: STREAM_SENDER[u Not tainted 4.18.0-13-generic #14~18.04.1-Ubuntu
  37 Jul 30 06:30:56 userv1 kernel: [19816692.905294] Hardware name: Micro-Star International Co., Ltd. MS-7B79/X470 GAMING PLUS (MS-7B79), BIOS A.40 06/28/2018
  38 Jul 30 06:30:56 userv1 kernel: [19816692.905309] RIP: 0010:usercopy_abort+0x6d/0x80
  39 Jul 30 06:30:56 userv1 kernel: [19816692.905317] Code: 44 d0 53 48 c7 c0 50 3b d0 8e 51 48 c7 c6 b7 15 cf 8e 41 53 48 89 f9 48 0f 45 f0 4c 89 d2 48 c7 c7 40 3c d0 8e e8 46 1c e6 ff <0f> 0b 49 c7      c1 0a 43 d2 8e 4d 89 cb 4d 89 c8 eb a5 66 90 0f 1f 44
...
  49 Jul 30 06:30:56 userv1 kernel: [19816692.905427] Call Trace:
  50 Jul 30 06:30:56 userv1 kernel: [19816692.905435]  __check_object_size+0x15f/0x1a0
  51 Jul 30 06:30:56 userv1 kernel: [19816692.905444]  __probe_kernel_read+0x4c/0x90
  52 Jul 30 06:30:56 userv1 kernel: [19816692.905451]  bpf_probe_read+0x1d/0x50
```
We will avoid starting the plugin on this kernel.

Fixes #9634

##### Component Name
eBPF plugin

##### Test Plan
- Run the eBPF collector on a system with a non-listed kernel and check it runs.
- Add the non-affected kernel version to the `ebpf_kernel_reject_list.txt` file and verify that the collector refuses to run.
